### PR TITLE
Add timer for overlay to hide by themself

### PR DESCRIPTION
**Pull request by Hpn4**

### CODE REVIEW

File: Holovibes/includes/gui/selections/interfaces/Overlay.hh [hunk 1]
```diff
@@ -62,16 +63,21 @@ class Overlay : protected QOpenGLFunctions
     Overlay(KindOfOverlay overlay, BasicOpenGLWindow* parent);
     virtual ~Overlay();
 
-    /*! \brief Get the zone selected */
-    const units::RectFd& getZone() const;
-
     /*! \brief Get the kind of overlay */
-    const KindOfOverlay getKind() const;
+    const KindOfOverlay getKind() const { return kOverlay_; }
```
Comment (ArthurCourselle) [inline Holovibes/includes/gui/selections/interfaces/Overlay.hh:67]: Put `inline` for more clarity

File: Holovibes/includes/gui/selections/interfaces/Overlay.hh [hunk 2]
```diff
@@ -62,16 +63,21 @@ class Overlay : protected QOpenGLFunctions
...
     /*! \brief Return if the overlay have to be deleted */
-    const bool isActive() const;
+    const bool isActive() const { return active_; }
+
+    /*! \brief Get the time before the overlay will hide */
+    const QDateTime getTimeBeforeHide() const { return time_before_hide_; }
+
+    /*! \brief Set the time before the overlay will hide */
+    void setTimeBeforeHide(QDateTime time) { time_before_hide_ = time; }
+
```
Comment (ArthurCourselle) [inline Holovibes/includes/gui/selections/interfaces/Overlay.hh:80]: Same as upper `inline` keyword is more understandable even if .hh functions are inlined

File: Holovibes/includes/gui/selections/overlay_manager.hh [hunk 3]
```diff
@@ -20,18 +22,19 @@ class OverlayManager
...
-    void create_default();
+    void enable(bool make_current = true, int ms = -1);
 
     /*! \brief Disable all the overlay of kind ko */
-    bool disable_all(KindOfOverlay ko);
-    /*! \brief Enable all the overlay of kind ko */
-    bool enable_all(KindOfOverlay ko);
-    /*! \brief Disable all the overlays. If def is set, it will create a default overlay. */
-    void reset(bool def = true);
+    bool disable(KindOfOverlay ko);
```
Comment (ArthurCourselle) [inline Holovibes/includes/gui/selections/overlay_manager.hh:34]: In doc add \return for doxygen

**Review by ArthurCourselle — COMMENTED** [review]
Just change the 2-3 mentionned things and good for me

File: Holovibes/includes/gui/selections/interfaces/Overlay.hh [hunk 4]
```diff
@@ -136,6 +142,9 @@ class Overlay : protected QOpenGLFunctions
     /*! \brief Transparency of the overlay, between 0 and 1 */
     float alpha_;
 
+    /*! \brief The time in ms when the overlay will dissappear */
```
Comment (Baschtien) [inline Holovibes/includes/gui/selections/interfaces/Overlay.hh:145]: Disappear* (sorry)

File: Holovibes/sources/API.cc [hunk 5]
```diff
@@ -1500,19 +1499,18 @@ void display_reticle(bool value)
     set_reticle_display_enabled(value);
 
     if (value)
-    {
-        UserInterfaceDescriptor::instance().mainDisplay->getOverlayManager().create_overlay<gui::Reticle>();
-        UserInterfaceDescriptor::instance().mainDisplay->getOverlayManager().create_default();
-    }
+        UserInterfaceDescriptor::instance().mainDisplay->getOverlayManager().enable<gui::Reticle>(false, 500);
```
Comment (Baschtien) [inline Holovibes/sources/API.cc:1502]: I feel like 500ms is a bit too short, for me 1s would be better
But if michael asked for it or something, its still just fine

**Review by Baschtien — APPROVED** [review]

**Review by TitouanGragnic — APPROVED** [review]

**Pull request merged**